### PR TITLE
Teach EdDSA signature algorithms about AlgorithmIdentifiers

### LINKED ISSUE #11875: OpenSSL 3.0.0-alpha2 EdDSA X509 signing
In OpenSSL 3.0.0-alpha2 ed25519 and ed448 keys fail when trying to use them to sign with `X509_sign`, `X509_CRL_sign`, and `X509_REQ_sign`. Based on some of the conversation in https://github.com/openssl/openssl/issues/11845 this could also possibly be an instance of a function not understanding provider side keys yet, but that is pure speculation on my part.

No errors are added to the stack, but the functions return 0.

Calling `EVP_PKEY_id` on ed25519/ed448 public keys returned from `X509_get_pubkey` also returns a 0 when we expect `NID_ED25519` (or the equivalent for ed448)

### PR DESCRIPTION
**Pull request by mattcaswell**
The other signature algorithms know how to create their own
AlgorithmIdentifiers, but the EdDSA algorithms missed this.

Fixes #11875

### CODE REVIEW

**Review by slontis — COMMENTED** [review]
Looks ok to me..
Need to update EVP_SIGNATURE-ED25519.pod. (See EVP_SIGNATURE-ECDSA.pod)

**Comment by slontis** [comment]
Travis error is relevant.

**Comment by mattcaswell** [comment]
> Need to update EVP_SIGNATURE-ED25519.pod. (See EVP_SIGNATURE-ECDSA.pod)

Documentation added. Also a fixup pushed to address the travis failure.

**Review by t8m — APPROVED** [review]
LGTM

**Comment by mattcaswell** [comment]
Pushed. Thanks.

File: providers/common/der/der_ecx_key.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+
+#include <openssl/obj_mac.h>
+#include "internal/packet.h"
+#include "prov/der_ecx.h"
+
+int DER_w_algorithmIdentifier_X25519(WPACKET *pkt, int cont, ECX_KEY *ec)
+{
+    return DER_w_begin_sequence(pkt, cont)
+        /* No parameters (yet?) */
```
Comment (kaduk) [inline providers/common/der/der_ecx_key.c:17]: FWIW I would be very surprised if parameters were defined in the future.
A major design point/goal was to have these be simple, with no parameters, with inputs/outputs as fixed-length byte strings.

**Pull request closed without merging**